### PR TITLE
Runner::processFile(): make internal error more informative

### PR DESCRIPTION
**Pull request by jrfnl**
When sniffs use traits/utility classes/helper functions, the message for the `Internal.Exception` can be unclear as it may point to the trait/utility classes/helper function as the culprit without any indication of which sniff caused the error.

This can cause bugs in sniffs to be reported by end-users in the wrong repository, but even when reported in the right repository, it makes the report unclear and means the maintainer will need to do more work to try and reproduce the issue.

This PR attempts to make the `Internal.Exception` error message more informative by pinpointing the sniff which caused the issue to occur. While this _may_ result in duplicate information when no traits/utility classes/helper functions are involved, for those cases when those are involved, I believe this will be a useful change.

Example output for a (simulated) error _without_ this change:
```
------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: Undefined variable
   |       | $tabWidth in
   |       | path/to/PHP_CodeSniffer/src/Standards/Generic/Sniffs/WhiteSpace/DisallowSpaceIndentSniff.php
   |       | on line 64 (Internal.Exception)
------------------------------------------------------------------------------------------------------------------------
```

Example output for the same (simulated) error _with_ this change:
```
------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: Undefined variable
   |       | $tabWidth in
   |       | path/to/PHP_CodeSniffer/src/Standards/Generic/Sniffs/WhiteSpace/DisallowSpaceIndentSniff.php
   |       | on line 64
   |       | The error originated in the Generic.WhiteSpace.DisallowSpaceIndent sniff on line 64.
   |       | (Internal.Exception)
------------------------------------------------------------------------------------------------------------------------
```

Example output for a (simulated) error _without_ this change for a sniff using a helper function:
```
------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: The
   |       | PHPCSUtils\Tokens\Collections::functionDeclarationTokensBC() method is deprecated since PHPCSUtils
   |       | 1.0.0-alpha4. Use the PHPCSUtils\Tokens\Collections::functionDeclarationTokens() method instead. in
   |       | path/to/PHPCSUtils/PHPCSUtils/Tokens/Collections.php on line 753
   |       | (Internal.Exception)
------------------------------------------------------------------------------------------------------------------------
```

Example output for the same (simulated) error _with_ this change for a sniff using a helper function:
```
------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted. The error message was: The
   |       | PHPCSUtils\Tokens\Collections::functionDeclarationTokensBC() method is deprecated since PHPCSUtils
   |       | 1.0.0-alpha4. Use the PHPCSUtils\Tokens\Collections::functionDeclarationTokens() method instead. in
   |       | path/to/PHPCSUtils/PHPCSUtils/Tokens/Collections.php on line 753
   |       | The error originated in the PHPCompatibility.Interfaces.NewInterfaces sniff on line 201.
   |       | (Internal.Exception)
------------------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

File: src/Runner.php [hunk 1]
```diff
@@ -652,6 +652,39 @@ public function processFile($file)
...
+                if (isset($step['file']) === false) {
+                    continue;
+                }
+
+                if (substr($step['file'], -9) === 'Sniff.php') {
+                    $sniffStack = $step;
+                    continue;
+                }
+
+                if (empty($sniffStack) === false) {
```
Comment (BafS) [inline src/Runner.php:669]: why not?
```suggestion
                if (!empty($sniffStack)) {
```

or even more simply

```suggestion
                if ($sniffStack) {
```

both are more readable

Comment (BafS) [reply src/Runner.php:669]: Oh I see, I was not aware of this sorry, it's a bit strange because it's less readable and less efficient.

**Review by BafS — APPROVED** [review]

File: src/Runner.php (hunk 1 above)
Comment (jrfnl) [reply src/Runner.php:669]: 🤷🏻‍♀️

**Comment by jrfnl** [comment]
Found a logic error in the original code. Fixed it and force pushed the commit again.

**Comment by jrfnl** [comment]
FYI: this fix is included in today's [PHP_CodeSniffer 3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).

**Pull request merged**